src/coreclr/src/tools/aot/crossgen2/Program.cs

```diff
@@ -347,6 +348,27 @@ private int Run()
                         }
                     }
 
+                    foreach (string inputFilePath in allInputFilePaths.Values)
+                    {
+                        EcmaModule module = typeSystemContext.GetModuleFromPath(inputFilePath);
+                        foreach (MetadataType type in module.GetAllTypes())
+                        {
+                            foreach (EcmaMethod method in type.GetAllMethods())
```

Comment:
is this iteration over all methods lead to further startup concerns?

Reply:
Removed in 3rd commit.